src/mudlet-lua/lua/DebugTools.lua

```diff
@@ -376,3 +376,64 @@ function display(...)
     echo((inspect(arg[1]) or 'nil') .. '\n')
   end
 end
+
+local errc
+-- leave errorc in the global table if and only if this is the mudlet self-test profile for running Busted tests
+-- this is because we need to spy on it to for testing.
+if getProfileName() ~= "Mudlet self-test" then
+  errc = errorc
+  _G.errorc = nil       -- and set to nil since it is internal only for the following functions.
+end
+-- undocumented, internal function
+local function printX(options)
+  local errorc = errc and errc or errorc
+  local func = options.func or debugc
+  local showTrace = options.showTrace
+  local msg = options.msg or ""
+  local halt = options.halt
+  local stackTable = debug.traceback():gsub("\t", "  "):gsub("%[string ",""):split("\n")
```

Comment:
Could it be optimized to only retrieve the traceback if it's needed? Afaik it's not a cheap function

Reply:
if I can find another way of reliably determining what level of the stack to inspect for the function information, then yes. But I was not able to find that when I went looking. That being said, in testing just now it only took 0.022 milliseconds to run 1000 printDebug statements, so I'm not sure at 0.022microseconds per printDebug it's adding significant overhead?

Reply:
![image](https://user-images.githubusercontent.com/3660/135463895-ad26939f-27da-4c3f-87a7-4573274349d2.png)

```lua
local epoch = getEpoch()
for i=1,1000 do
  printDebug("This is a test!")
end
local totalTime = getEpoch() - epoch
cecho(f"Took {totalTime} seconds to run 1000 printDebugs")
```
If you want to repeat the test